app/jobs/subscription_placement_job.rb

```diff
@@ -19,10 +20,15 @@ def summarizer
   end
 
   def proxy_orders
+    # Order cycles become active at certain times. We need to cache the
+    # collection here to keep the result deterministic and avoid dead locks.
+    @active_order_cycle_ids ||= OrderCycle.active.pluck(:id)
+
     # Loads proxy orders for open order cycles that have not been placed yet
     ProxyOrder.not_canceled.where(placed_at: nil)
-      .joins(:order_cycle).merge(OrderCycle.active)
+      .where(order_cycle_id: @active_order_cycle_ids)
```

Comment:
> We need to cache the collection here to keep the result deterministic

This `proxy_orders` method is only called once in the entire lifecycle of the job though, right? And `@active_order_cycle_ids` is only called once. I don't understand in what way it's actually being cached or how it's practically different from what was here before...? :thinking:

Again; just curious. :smile:

Reply:
Partly I'm asking because this will trigger an extra query that plucks the ID of every active OrderCycle and assign it to the `@active_order_cycle_ids` ivar, but the previous version would avoid doing that...

Reply:
> This proxy_orders method is only called once in the entire lifecycle of the job though, right?

Nope. The way I wrote it, the method gets called in the while loop once per proxy order to process. I added it so that repeated queries would not change with time and when calling `first` we would always get the next unprocessed proxy order.

But I'm thinking about querying them all in one batch again. The issue is the transaction though. I'll comment below.